code/entities/machine/soda-fountain/SodaFountain.FillableCup.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Sandbox;
+
+namespace Cinema;
+
+public partial class FillableCup : ModelEntity
+{
+    public Dispenser Dispenser { get; set; }
+    public string ItemId { get; set; } 
+    public bool IsAssembled { get; set; }
+
+    /// <summary>
+    /// 
+    /// </summary>
+    public override void Spawn()
+    {
+        base.Spawn();
+
+        SetModel("models/papercup/walker/sodafountaincup/papercup.vmdl");
+
+        SetupPhysicsFromModel(PhysicsMotionType.Static);
+    }
+    /// <summary>
+    /// 
+    /// </summary>
+    public void Initialize()
+    {
+        SetMaterialGroup((int)SodaFountain.GetCupColorBySodaType(Dispenser.SodaType));
+
+        Dispenser.SetCup(this);
+    }
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <returns> </returns>
+    public void Assemble()
+    {
+        //Set to "assembled" body group
```

Comment:
Clean State Nit: instant return if `IsAssembled` is already true.

Reply:
Added instant return if IsAssembled is true